client/grpc_client.rs

```diff
@@ -158,13 +160,22 @@ pub async fn spawn_grpc_client(
     let grpc_host = grpc_url.host();
     info!("Streaming from gRPC server: {:?}", grpc_host);
     let mut client = MTransactionClient::new(channel);
-    mtx_stream(
+    let mx = mtx_stream(
         grpc_host.unwrap_or("unknown").to_string(),
         &mut client,
         tx_transactions,
         metrics,
     )
     .await;
 
+    match mx {
+        Ok(_) => {
+            info!("Stream ended successfully");
+        }
+        Err(e) => {
+            return Err(e);
+        }
+    }
+
     Ok(())
```

Comment:
```rust
    mtx_stream(
        grpc_host.unwrap_or("unknown").to_string(),
        &mut client,
        tx_transactions,
        metrics,
    )
    .await
    .inspect(|_| info!("Stream ended successfully"))
```

Reply:
This throws `use of unstable library feature 'result_option_inspect'` for the `.inspect()` call

I'd say let's not use anything unstable. The whole thing is unstable enough already

Reply:
ah, this is in the stable from 1.76.0 https://doc.rust-lang.org/std/result/enum.Result.html#method.inspect
I think we are okay to bump rust to 1.76.0 (in separate PR)

Reply:
if we want to stay with 1.70.0 we can use map
```rust
.map(|v| {
  info!("Stream ended successfully");
  v
})
```